README.md

```diff
@@ -77,14 +79,55 @@ Usage
 ```
 16. Click “Apply Policy”.
 
+## Generating a proper IAM role
+Note : Only applies when you want to access Device Farm through an IAM Role.
+
+You need to create one Device Farm access role to access the Device Farm Resource and one User or Role depending upon how you are running Jenkins.
+
+- Creating Device Farm Role.
+1. Log into your AWS web console UI.
+2. Click “Identity & Access Management”.
+3. On the left-hand side of the screen, click “Roles”.
+4. Click “Create Role”.
+5. In the AWS service section select Ec2 and click on Next.
+6. Under the Inline Policies header, search and click the “AWSDeviceFarmFullAccess ” policy.This will give complete device farm access.
+7. Now we need to update the maximum expiration of the credentials to 4hrs.
+8. Go the the summary page of the role and click on edit next to "Maximum CLI/API session duration" and select 4hrs.
+9. Now you want to give the access to the role/user to assume this role
+Open the new role you have created in console and then click on "Trust Relationships" tab.Click on "Edit Trust Relationship" and enter the following policy.
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "To be replace by Arn of user/role running on the Jenkins machine"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+
+- Setting up Role/User running Jenkins
+
+1. We need to also give access 
+2. Go to the user or role that you want to add policy to.
+3. In the permission tab click on Create Policy.
+4. Select STS in the service name, search for assume role in actions and select it.
+5. In the Resources section enter the arn for the device farm role.
+6. Click on review policy and then enter name and description and click on Create policy.
+
 ## First-time configuration instructions:
 
 1. Log into your Jenkins web UI.
 2. On the left-hand side of the screen, click “Manage Jenkins”
 3. Click “Configure System”.
 4. Scroll down to the “AWS Device Farm” header.
-5. Copy/paste your AKID and SKID you created previously into their respective boxes.
-6. Click “Save”.
+5. If you are using I AM Role to configure the Jenkins Plugin, add the arn for the role here.
```

Comment:
5. If you are using an IAM Role to configure the Jenkins Plugin, add the ARN for the role here.

Reply:
Changed